client/src/pages/ProfileListings/ProfileListings.tsx

```diff
@@ -0,0 +1,164 @@
+
+  const moment = extendMoment(Moment);
+
+  const handleShowMore = () => {
+    const numberOfUsers = displayedUsers.length;
+    const lastUser = users[displayedUsers.length];
+    if (lastUser === undefined) {
+      setSnackbarOpen(true);
+    }
+    setDisplayedUsers(users.slice(0, numberOfUsers + 3));
+  };
+
+  const handleSnackbarClose = (event?: React.SyntheticEvent, reason?: string) => {
+    if (reason === 'clickaway') {
+      return;
+    }
+    setSnackbarOpen(false);
+  };
+
+  const handleCalendarOpen = () => {
+    setCalendarOpen((prevState) => !prevState);
+  };
+
+  const updateDateRange = (dateFrom: Date | null, dateTo: Date | null) => {
+    if (dateFrom instanceof Date && dateTo instanceof Date) {
+      setDateRange({
+        from: dateFrom,
+        to: dateTo,
+      });
+      const formattedDateFrom = moment(dateFrom).format('MMMM Do YYYY');
+      const formattedDateTo = moment(dateTo).format('MMMM Do YYYY');
+      setFormattedDateRange(`${formattedDateFrom} – ${formattedDateTo}`);
+    } else {
+      setFormattedDateRange(`any`);
+    }
+  };
+
+  const updateSearch = () => {
+    const searchResults: User[] = [];
+    setDisplayedUsers([]);
+    users.map((user: User) => {
+      if (user.city.toLowerCase().includes(search)) {
+        searchResults.push(user);
+        setDisplayedUsers(searchResults);
+      }
+    });
+  };
+
+  //   Updates based on date selection
+  useEffect(() => {
+    if (formattedDateRange === 'any') {
+      setDisplayedUsers(users.slice(0, 6));
+    } else {
+      if (dateRange.from && dateRange.to) {
+        const searchResults: User[] = [];
+        setDisplayedUsers([]);
+        const range = moment.range(dateRange.from, dateRange.to);
+        users.map((user) => {
+          //   If at least one available date falls within the selected range, the user
+          //   is added to the displayedUsers array.
+          const isInDateRange = (date: Date) => {
+            return range.contains(date);
+          };
+          if (user.availableDates.some(isInDateRange)) {
+            searchResults.push(user);
+            setDisplayedUsers(searchResults);
+          }
+        });
+      }
+    }
+  }, [formattedDateRange, setFormattedDateRange]);
+
+  //   Updates based on search parameters
+  useEffect(() => {
+    if (search) {
+      updateSearch();
+    } else {
+      setDisplayedUsers(users.slice(0, 6));
+    }
+  }, [search, setSearch]);
+
+  const profileCardGrid = (
+    <Grid container className={classes.profilesContainer} xs={9}>
+      {displayedUsers.map((user) => (
+        <ProfileCard user={user} key={user._id} />
+      ))}
+      <Snackbar open={snackbarOpen} autoHideDuration={2000} onClose={handleSnackbarClose}>
+        <Alert severity="info">No more users found</Alert>
+      </Snackbar>
+    </Grid>
+  );
+
+  return (
+    <Grid container className={classes.root}>
+      <Typography variant="h4" className={classes.title}>
```

Comment:
```suggestion
      <Typography variant="h4" component="h1" className={classes.title}>
```
There should be atleast one `h1` heading in the page followed by other heading. By default, material-ui, make the tag same as the variant provided. [https://www.w3.org/WAI/test-evaluate/preliminary/#headings](https://www.w3.org/WAI/test-evaluate/preliminary/#headings)